spec/topological_inventory/sync/inventory_upload/validator_worker_spec.rb

```diff
@@ -13,7 +13,10 @@
       \"url\":\"/tmp/upload/schema.tar.gz\"}"
     end
 
-    before { expect(validator).to receive(:parse_inventory_payload).and_return(inventory) }
+    before do
+      expect(TopologicalInventory::Sync::InventoryUpload::Parser)
+        .to receive(:open_url).and_yield(targz(inventory))
```

Comment:
@gtanzillo this now also tests the parse_inventory_payload method by yielding the raw tgz file string instead of just returning the inventory hash